src/orders/order_modules/data_mapper/order_mapper.py

```diff
@@ -7,49 +7,135 @@
+        self.logger = Logger()
+        self.location_service = location_service or Geolocation()
+
+    def fetch_geolocation(self) -> Dict[str, float]:
+        """
+        Fetches geolocation data based on the order data's delivery address.
+
+        :return: A dictionary with latitude and longitude
+        :rtype: Dict[str, float]
+        """
+        geolocation = self.order_data.get("geolocation", None)
+        if geolocation is None:
+            self.logger.info(
+                "Input did not include geolocation data, invoking Geolocation Service")
+            geolocation = self.location_service.get_lat_and_long_from_street_address(
+                str_address=self.order_data.get("delivery_address")
+            )
+            return geolocation
+        else:
+            self.logger.info("Using provided geolocation data from input")
+            return geolocation
+
+    def get_available_driver(self,
+                             geolocation: Dict[str, float],
+                             delivery_time: str,
+                             delivery_date: str):
+
+        customer_location = (geolocation.get("latitude"),
+                             geolocation.get("longitude"))
+
+        if delivery_date is None:
+            delivery_date = self.order_data.get(
+                "delivery_date", datetime.now().strftime("%Y-%m-%d"))
+
+        dao = OrderDAO()
+        orders = dao.fetch_orders(
+            primary_key=ORDERS_PRIMARY_KEY,
+            query_value=delivery_date
+        )
+        orders = orders["payload"]
+
+        planner = DeliveryScheduler()
+        driver = planner.assign_driver_for_delivery(
+            customer_location=customer_location,
+            delivery_time=delivery_time,
+            order_date=delivery_date,
+            orders=orders
+        )
+        if driver:
+            return driver
+        else:
+            raise BusinessError("No drivers available")
 
     def build_order(
         self,
-        order_data: Dict[str, Any],
         username: str,
-        geolocation: Dict[str, float],
-        errors: List[Dict[str, str]]
+        uid: str = None
     ) -> Dict[str, Any]:
         """This function will create a dictionary to send to DynamoDB to create a new record
 
         Arguments:
-            order_data -- Processed clients input
             username -- who is sending the request
-            geolocation -- Lat and Long
+            uid -- order unique identifier
 
         Returns:
             Object needed by DynamoDB to create a record
         """
+        order_errors = []
+
+        if uid is None:
+            uid = str(uuid.uuid4())
 
-        items = [item.__dict__ for item in order_data.get("cart_items", [])]
-        status = OrderStatus.ERROR.value if errors else OrderStatus.CREATED.value
+        geolocation = self.fetch_geolocation()
+        if geolocation is None:
+            # TODO Discuss: No order should be created without geolocation.
```

Comment:
Mesa de control or Admins will get a warning, and cant schedule any order until they fix this, location service is not 100% accurate, specially with Cotos or new buildings,thats why geolocation service can return something innacurate and they might need to call the customer to validate the address using google maps or something similar